app/src/main/java/com/team41/boromi/adapters/PagerAdapter.java

```diff
@@ -0,0 +1,41 @@
+package com.team41.boromi.adapters;
+
+import android.os.Bundle;
+import android.util.Pair;
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+import androidx.lifecycle.Lifecycle;
+import androidx.viewpager2.adapter.FragmentStateAdapter;
+import com.team41.boromi.FragmentFactory;
+import java.util.ArrayList;
+
+public class PagerAdapter extends FragmentStateAdapter {
+
+  private ArrayList<Pair<Class<? extends Fragment>, Bundle>> fragmentClasses;
+
+  public PagerAdapter(@NonNull FragmentManager fragmentManager,
+      @NonNull Lifecycle lifecycle,
+      ArrayList<Pair<Class<? extends Fragment>, Bundle>> fragmentClasses) {
+    super(fragmentManager, lifecycle);
+    this.fragmentClasses = new ArrayList<>(fragmentClasses);
+  }
+
+  @NonNull
+  @Override
+  public Fragment createFragment(int position) {
+    Pair<Class<? extends Fragment>, Bundle> p = fragmentClasses.get(position);
+    try {
+      return FragmentFactory.createFragment(p.first, p.second);
+    } catch (InstantiationException | IllegalAccessException e) {
```

Comment:
If this happens, then the app will continue. Is that desired and how is it handled ? Should you throw some kind of runtime exception if the fragment fails to get created with these exceptions ?
What are some ways that FragmentFactory will throw InstantiationException

Reply:
FragmentFactory will throw InstantiationException if it fails to create the class with newInstance.

Reply:
Right and but the app is going to continue if it does, it's still going to return null. That means the catch just catches the error and does nothing, is that okay ?